### PR TITLE
[New] Add logdown.js w/ npm auto-update

### LINKED ISSUE #10272: [Request] Add "Logdown"
**Library name:** Logdown 
**Git repository url:** https://github.com/caiogondim/logdown.js
**npm package url(optional):** https://www.npmjs.com/logdown
**License(s):** MIT
**Official homepage:** http://caiogondim.github.io/logdown.js/
**Wanna say something? Leave message here:** Loving your service!

=====================
Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster than just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/40881508-request-add-logdown?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by sashberd**
close #10272, cc @caiogondim

Pull request for issue: #
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/caiogondim/logdown.js
 * Official website (optional, not the repository): http://caiogondim.github.io/logdown.js/
 * NPM package url (optional): https://www.npmjs.com/package/logdown
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 29
   - Count of stars: 881
   - Count of forks: 32
 * NPM download stats (optional):
   - Downloads in the last day: 185
   - Downloads in the last week: 1566
   - Downloads in the last month: 4925

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/logdown.js/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "console",
+    "info",
+    "error",
+    "logging"
+  ],
+  "author": "Caio Gondim <me@caiogondim.com> (https://caiogondim.com)",
+  "homepage": "http://caiogondim.github.io/logdown.js/",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/brightcove/videojs-thumbnails.git",
```
Comment (sufuf3) [inline ajax/libs/logdown.js/package.json]: The commit message said use npm auto-update, but this is git auto-update config.
Which one is correct?
Also, `target` is wrong.
Please modify the correct auto-config.
Many thanks.

File: ajax/libs/logdown.js/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "fileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "**/logdown*.(js|map)"
```
Comment (sufuf3) [inline ajax/libs/logdown.js/package.json]: We use exclude mode when the basePath is `dist`.
Let's use exclude mode.
Thanks.

File: ajax/libs/logdown.js/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "log",
+    "markdown",
+    "debug",
+    "console",
+    "info",
+    "error",
+    "logging"
+  ],
+  "author": "Caio Gondim <me@caiogondim.com> (https://caiogondim.com)",
+  "homepage": "http://caiogondim.github.io/logdown.js/",
```
Comment (sufuf3) [inline ajax/libs/logdown.js/package.json]: Let's use `https`.
Thanks.

File: ajax/libs/logdown.js/package.json (hunk 2 above)
Comment (sashberd) [reply ajax/libs/logdown.js/package.json]: I little stuck with this pattern. Could you help to to figure put what it should be. Thanks

**Comment by sashberd** [comment]
@sufuf3 The PR was updated. FYR

File: ajax/libs/logdown.js/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  "author": "Caio Gondim <me@caiogondim.com> (https://caiogondim.com)",
+  "homepage": "https://caiogondim.github.io/logdown.js/",
+  "autoupdate": {
+    "source": "git",
+    "target": "https://github.com/caiogondim/logdown.js.git",
+    "fileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "*.(js|map)"
```
Comment (extend1994) [inline ajax/libs/logdown.js/package.json]: Exclude unnecessary files like `**/!(*.gzip|*.ts)`

**Comment by sashberd** [comment]
@extend1994 the PR was updated FYR

**Comment by sashberd** [comment]
@sufuf3 You are right, My apologies for this inconsistent content. I change config to npm and add manually versions that are not exists on npm. FYR

**Comment by sashberd** [comment]
@sufuf3 @extend1994 Ping

File: ajax/libs/logdown.js/package.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "logdown.js",
```
Comment (extend1994) [inline ajax/libs/logdown.js/package.json]: Because you used npm auto-update, using npm package name might be a better idea?

**Comment by sashberd** [comment]
@extend1994 The PR was updated. FYR

**Comment by sashberd** [comment]
@extend1994 The PR was updated . FYR

**Comment by mercyg** [comment]
LGTM

**Comment by sashberd** [comment]
@xtuc FYR

**Pull request merged**